Project/src/MakeCall/RemoteParticipantCard.js

```diff
@@ -18,25 +19,42 @@ export default class RemoteParticipantCard extends React.Component {
         };
     }
 
-    componentWillMount() {
+    componentWillUnmount() {
+        this.remoteParticipant.off('isMutedChanged', () => {});
+        this.remoteParticipant.off('stateChanged', () => {});
+        this.remoteParticipant.off('isSpeakingChanged', () => {});
+        this.remoteParticipant.off('displayNameChanged', () => {});
+        this._isComponentMounted = false;
+    }
+
+    componentDidMount() {
         this.remoteParticipant.on('isMutedChanged', () => {
-            this.setState({ isMuted: this.remoteParticipant.isMuted });
-            if (this.remoteParticipant.isMuted) {
-                this.setState({ isSpeaking: false });
+            if (this._isComponentMounted){
```

Comment:
This check doesnt seem right. I shouldnt have to check this flag everytime to make the update

Reply:
the problem is that those listeners keep getting called and after the component and it was causing some warnings related to memory in console